pyblish_bumpybox/plugins/ftrack/extract_ftrack_components.py

```diff
@@ -47,51 +50,42 @@ def process(self, instance):
 
     def add_ftrack_components(self, instance, component_path):
 
+        component = {
+            "assettype_data": {
+                "short": instance.data.get("assettype_short", "upload")
+            },
+            "asset_data": {
+                "name": instance.data.get(
+                    "asset_name",
+                    instance.context.data["ftrackTask"]["name"]
+                ),
+                "parent": instance.data.get(
+                    "asset_parent",
+                    instance.context.data["ftrackTask"]["parent"]
+                )
+            },
+            "assetversion_data": {
+                "version": instance.data.get(
+                    "version", instance.context.data["version"]
+                )
+            },
+            "component_data": {
+                "name": instance.data.get(
+                    "component_name", instance.data["name"]
+                ),
+            },
+            "component_path": component_path,
+            "component_overwrite": instance.data.get(
+                "component_overwrite", True
+            ),
+            "component_location": instance.data.get("component_location", None)
```

Comment:
Initially ```component_location``` is not set anywhere in pyblish_bumpybox plugins. It is up to the users to customize this.
If ```component_location``` is not present in ```instance.data``` then it will not add ```component_location``` to the component, which results in ```session.pick_location()``` in https://github.com/pyblish/pyblish-ftrack/blob/master/pyblish_ftrack/plugins/integrate_ftrack_api.py#L156
So to setup a location for all components, you just need to have it configured on launch, or make a studio specific extractor that adds the locations for all components.